sensibo/Utils.js

```diff
@@ -78,46 +91,51 @@ module.exports = (device, platform) => {
 			const validValues = characteristic.props.validValues
 
 			if (Number.isNaN(newValue)) {
-				// non-number is valid for many usecases
+				// non-number is valid for many use cases
 				// TODO: could check if props.format is float or int, then compare and fail if needed?
-				log.easyDebug(`${device.name} - '${newValue}' is not a number for characteristic ${characteristicName} (expected format '${format}') on service ${serviceName}... continuing`)
+				easyDebugError(`${device.name} - '${newValue}' is not a number for ${characteristicName} (expected format '${format}') for ${serviceName}... continuing`)
 			}
 
 			if (validValues && !validValues.includes(newValue)) {
-				log.easyDebug(`${device.name} - '${newValue}' not in validValues: ${validValues} for characteristic ${characteristicName} on service ${serviceName}... skipping update`)
+				easyDebugError(`${device.name} - '${newValue}' not in validValues: ${validValues} for ${characteristicName} for ${serviceName}... skipping update`)
 
 				return
 			}
 
-			// TODO: CurrentTemperature value being returned seems to need rounding?
-			// e.g. "22.60000000000001"
-
+			// NOTE: CurrentTemperature value  being returned (e.g. "22.60000000000001") needs rounding!
 			if (minStep) {
-				const roundedValue = minStep < 1 ? Math.round((newValue + Number.EPSILON) * 10) / 10 : Math.round(newValue + Number.EPSILON)
-
-				if (roundedValue !== newValue) {
-					log.easyDebug(`${device.name} - '${newValue}' doesn't meet the rounding requird by minStep: ${minStep} for characteristic ${characteristicName} on service ${serviceName}... rounding to ${roundedValue}`)
-					newValue = roundedValue
+				if (typeof(newValue) == 'number') {
+					const newValuePlusEpsilon = newValue + Number.EPSILON
+					const roundedValue = minStep < 1 ? Math.round((newValuePlusEpsilon) * 10) / 10 : Math.round(newValuePlusEpsilon)
+
+					if (roundedValue !== newValue) {
+						easyDebugWarning(`${device.name} - '${newValue}' doesn't meet required rounding (minStep = ${minStep}) for ${characteristicName} for ${serviceName}, rounding: ${roundedValue}`)
+						newValue = roundedValue
+					}
+				} else{
+					easyDebugError(`${device.name} - '${newValue}' for ${characteristicName} for ${serviceName} is not of type 'number'... skipping update`)
+
+					return
```

Comment:
Not sure I understand your reasoning here? newValue must be of type number, we must return if not, otherwise we will try to compare newValue and maxValue/minValue which not make sense. Did I misunderstand your thinking?